### DISCUSSION TITLE
X-server does start automatically on clients

### DISCUSSION THREAD

**Discussion by mavrinac** [ltsp/ltsp#655] [Q&A]
After migrating from LTSP5 to TLSP21 on existing servers, when I boot a client, I am no longer presented with a greeter. I can login at the console and then run startx manually, so the X-server isn't starting automatically. How can do I get the graphical login working on client startup?

**Comment by intelliant01** [comment]
@mavrinac Do your clients have nvidia graphics?

**Reply by mavrinac** [reply]
Some do, but not the ones I'm asking about. They have radeon (AMD Kabini) with amdgpu capabilities.

**Comment by alkisg** [comment]
Which distribution and which kernel version?
Does a live USB stick with the exact same version work?
What kind of image, e.g. chrootless?
Do all the clients have this issue?
What is the output of `systemctl --failed` at that point (from the console)?

**Reply by mavrinac** [reply]
Debian Bullseye (11.2 )
linux-image-5.10.0-11-amd64
Yes, chrootless clients
I have about 110 of the same clients. I haven't tested the other ones (fewer of them) yet.
I have to get back to my office to get the list of failed services (It's not helping that I am trying to troubleshoot some of this remotely!). I will try to post that today.

**Comment by mavrinac** [comment]
My aplogies for taking so long to complete this discussion.
After looking more closely at the changes in the client setup and ltsp.conf, I have solved all of my problems with the upgrade. First of all, I kept post-init changes to the client confined to a single { POST_INIT_A=$(cat); } function definition and session/system service masking in ltsp.conf. I realized that the server configuration is much more closely mirrored in the client since LTSP 19 than it was in LTSP 5; that said, I included "sysetmctl set-default graphical.target" in the post-init code for the client - the server default was multi-user.target which was inherited by the client!
